platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -63,18 +64,69 @@
   private final int maximumShardsPerNode =
       NumberUtils.toInt(System.getProperty("solr.cloud.maxShardPerNode"), 2);
 
-  @Override
-  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
-    String zookeeperHosts = System.getProperty("solr.cloud.zookeeper");
+  private final Map<String, Integer> shardCountMap = new HashMap<>();
 
-    if (StringUtils.isBlank(zookeeperHosts)) {
+  private final Map<String, Integer> replicationFactorMap = new HashMap<>();
+
+  private final Map<String, Integer> maximumShardsPerNodeMap = new HashMap<>();
+
+  private String zookeeperHosts;
+
+  public SolrCloudClientFactory() {
+    String collectionList = System.getProperty("solr.cloud.collections");
+    if (StringUtils.isBlank(collectionList)) {
       LOGGER.warn(
-          "Cannot create Solr Cloud client without Zookeeper host list system property [solr.cloud.zookeeper] being set.");
-      throw new IllegalStateException("system property 'solr.cloud.zookeeper' is not configured");
+          "No solr.cloud.collections configuration found. Using default cloud configuration settings");
+    } else {
+      for (String collection : collectionList.split(",")) {
+        shardCountMap.put(
+            collection,
+            NumberUtils.toInt(System.getProperty("solr.cloud." + collection + ".shardCount"), 2));
+        replicationFactorMap.put(
+            collection,
+            NumberUtils.toInt(
+                System.getProperty("solr.cloud." + collection + ".replicationFactor"), 2));
+        maximumShardsPerNodeMap.put(
+            collection,
+            NumberUtils.toInt(
+                System.getProperty("solr.cloud." + collection + ".maxShardPerNode"), 2));
+      }
     }
+    zookeeperHosts = System.getProperty("solr.cloud.zookeeper");
+  }
+
+  @Override
+  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
+    checkConfig();
     LOGGER.debug(
         "Solr({}): Creating a Solr Cloud client using Zookeeper hosts [{}]", core, zookeeperHosts);
-    return new SolrClientAdapter(core, () -> createSolrCloudClient(zookeeperHosts, core));
+    SolrClientAdapter adaptor =
+        new SolrClientAdapter(core, () -> createSolrCloudClient(zookeeperHosts, core));
+    try {
+      if (!adaptor.isAvailable(30, 5, TimeUnit.SECONDS)) {
+        LOGGER.warn("Solr Client {} is not available after 30 seconds", core);
+      }
+    } catch (InterruptedException e) {
+      LOGGER.error("Unable to connect to solr client {}: {} ", core, e.getStackTrace());
```

Comment:
❗️ Improper way to handle interruptions. They need to be propagated up by re-interrupting the current thread.